src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/RestSenderBase.cs

```diff
@@ -13,18 +13,19 @@ public abstract class RestSenderBase
 
         private readonly int targetApiPort;
 
-        private readonly Uri publicationUri;
+        private Uri publicationUri;    
 
-        public RestSenderBase(ILoggerFactory loggerFactory, IFederationGatewaySettings settings, string route)
+        public RestSenderBase(ILoggerFactory loggerFactory, IFederationGatewaySettings settings)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.targetApiPort = settings.CounterChainApiPort;
-            this.publicationUri = new Uri(
-                $"http://localhost:{this.targetApiPort}/api/FederationGateway/{route}");
         }
 
-        protected async Task SendAsync<T>(T model)
+        protected async Task SendAsync<T>(T model, string route)
         {
+            this.publicationUri = new Uri(
```

Comment:
If we are overwriting this with every send why does it need to be a member?

Reply:
Ha - no it doesn't I was going to remove it, but worried about changing something not related to this PR - I'm happy to change now if that's okay.